nginx.conf

```diff
@@ -35,12 +35,25 @@ http {
 
     <% require 'json' %>
     <% proxies = JSON.parse(ENV["PROXIES"] || "{}") %>
+    <%
+      secret = ENV["HMAC_SECRET"] ||
+        (JSON.parse(ENV["VCAP_SERVICES"])["user-provided"] || []).map { |service|
+          service["credentials"]["HMAC_SECRET"]
+        }.find { |secret|
+          secret
+        }
+    %>
 
     <% proxies.each do |path, route| %>
       location <%= path %> {
         resolver 8.8.8.8;
         set $backend "<%= route %>";
         proxy_pass $backend;
+        <% if secret %>
+        set_hmac_sha1 $signature "<%= secret %>" "$http_content_length\n$http_content_md5\n$http_content_type\n$http_date";
```

Comment:
Any need to include the request body in the signature string? At any rate, I'd add an extra newline to the end of the string in case you need to tack something else onto the end one day, since without the newline delimiter you could have two requests that hash to the same value when the end of the last header buts straight up against the beginning of the body.